pkg/cli/admin/catalog/build.go

```diff
@@ -0,0 +1,216 @@
+package catalog
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/operator-framework/operator-registry/pkg/apprclient"
+	"github.com/operator-framework/operator-registry/pkg/sqlite"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/klog"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/util/templates"
+
+	imgappend "github.com/openshift/oc/pkg/cli/image/append"
+)
+
+func NewBuildImageOptions(streams genericclioptions.IOStreams) *BuildImageOptions {
+	return &BuildImageOptions{
+		IOStreams:           streams,
+		AppRegistryEndpoint: "https://quay.io/cnr",
+		From:                "quay.io/operator-framework/operator-registry-server:latest",
+	}
+}
+
+func NewBuildImage(f kcmdutil.Factory, parentName string, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewBuildImageOptions(streams)
+	cmd := &cobra.Command{
+		Use:   "build",
+		Short: "Builds a registry container image from a collection operator manifests.",
+		Long: templates.LongDesc(`
+			Builds a registry container image from a collection operator manifests.
+
+			Extracts the contents of a collection of operator manifests to disk, and builds them into
+			an operator registry image.
+		`),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+	flags := cmd.Flags()
+
+	flags.StringVar(&o.From, "from", o.From, "The image to use as a base.")
+	flags.StringVar(&o.To, "to", o.To, "The Docker repository tag to upload the completed catalog image to.")
```

Comment:
`The image repository tag` instead of `Docker repository tag`? or `tag to apply to to built catalog image`? not sure.

Reply:
I updated to your suggestions, but I took these descriptions from `oc image append`, which this command delegates to.